semgrep/semgrep/commands/shouldafound.py

```diff
@@ -98,6 +93,16 @@ def shouldafound(
         sys.exit(2)
 
 
+def _get_git_email() -> Optional[str]:
+    try:
+        email = (
+            subprocess.check_output(["git", "config", "user.email"]).decode().strip()
+        )
+        return email
+    except Exception:
+        return None
```

Comment:
What you have definitely works. This may be a bit simpler:

```suggestion
def _get_git_email() -> str:
    ```
    :raises CalledProcessException: If not in a git environment / email not set
    ```
    return subprocess.check_output(["git", "config", "user.email"]).decode().strip()
```

Reply:
@nbrahms doesnt check_output raise an exception?

Reply:
addressed nathan's suggestion, resolving.